shared/chat/conversation/list-area/normal/index.desktop.js

```diff
@@ -3,94 +3,81 @@
+        return this.props.messageOrdinals.get(ordinalIndex)
+      }
+    },
   })
 
   _list: any
-  _keepIdxVisible: number = -1
-  _lastRowIdx: number = -1
-  // ScrolltoRow sometimes triggers load more. This'll all go away when we ditch react-virtualized. for now this fixes
-  // if you edit your last message that'l cause a load more sometimes due to scrolltop being 0 incorrectly
-  _ignoreScrollUpTill: number = 0
-
-  state = {
-    isLockedToBottom: true,
-    listRerender: 0,
-    selectedMessageKey: null,
-  }
-
-  componentDidUpdate(prevProps: Props, prevState: State) {
-    // Force a rerender if we passed a row to scroll to. If it's kept around the virutal list gets confused so we only want it to render once basically
-    if (this._keepIdxVisible !== -1) {
-      this.setState(prevState => ({listRerender: prevState.listRerender + 1})) // eslint-disable-line react/no-did-update-set-state
-      this._keepIdxVisible = -1
-    }
-    this._lastRowIdx = -1 // always reset this to be safe
+  // If we should stick to the bottom
+  _isLockedToBottom: boolean = true
 
+  componentDidUpdate(prevProps: Props) {
     if (this.props.editingOrdinal && this.props.editingOrdinal !== prevProps.editingOrdinal) {
       const idx = this.props.messageOrdinals.indexOf(this.props.editingOrdinal)
       if (idx !== -1) {
-        this._ignoreScrollUpTill = Date.now() + 1000
-        this._list && this._list.scrollToRow(idx)
+        this._list && this._list.scrollToRow(idx + 1)
+      }
+    } else if (this.props.messageOrdinals.size !== prevProps.messageOrdinals.size && this._list) {
+      // try and maintain scroll position, doens't work great
+      if (prevProps.messageOrdinals.size > 1) {
+        const toFind = prevProps.messageOrdinals.first()
+        if (toFind === this.props.lastLoadMoreOrdinal) {
+          const idx = toFind ? this.props.messageOrdinals.indexOf(toFind) : -1
+          if (idx !== -1) {
+            const scrollToIdx = idx + 1
+            this._list.scrollToRow(scrollToIdx)
+          }
+        }
       }
     }
   }
 
   componentWillReceiveProps(nextProps: Props) {
     if (this.props.conversationIDKey !== nextProps.conversationIDKey) {
       this._cellCache.clearAll()
-      this.setState({isLockedToBottom: true})
+      this._isLockedToBottom = true
     }
 
     if (this.props.messageOrdinals.size !== nextProps.messageOrdinals.size) {
-      if (this.props.messageOrdinals.size > 1 && this._lastRowIdx !== -1) {
-        const toFind = this.props.messageOrdinals.get(this._lastRowIdx)
-        this._keepIdxVisible = toFind ? nextProps.messageOrdinals.indexOf(toFind) : -1
-      }
-      // Force the grid to throw away its local index based cache. There might be a lighterway to do this but
-      // this seems to fix the overlap problem. The cellCache has correct values inside it but the list itself has
-      // another cache from row -> style which is out of sync
-      this._cellCache.clearAll()
```

Comment:
I'm seeing overlap issues sometimes on desktop now, maybe caused by this?

Reply:
maybe, i was hoping we didn't need to throw all this data away but i'll put it back